torch/_dynamo/guards.py

```diff
@@ -1278,29 +1264,14 @@ def guard_fail_hook(
         # for everything else we just report the code that failed
 
         if isinstance(fail_reason, bool) and not fail_reason:
-            fail_reason = part
-        if isinstance(fail_reason, str):
-            reason += fail_reason
-            if config.report_all_guard_failures:
```

Comment:
https://github.com/pytorch/pytorch/blob/8768b87bd104cdf9be9be233045ca37e2f1e029f/torch/csrc/dynamo/eval_frame.c#L315-L316

@williamwen42 from our discussion on slack - I think you are mistaken that each of the above like `FUNCTION_MATCH` will generate a single guard function.

All of the guard checks get accumulated into a single `guard_fn`/`check_fn`

https://github.com/pytorch/pytorch/blob/f908b0e9a3b30fde12106a0396baa3d0e19507f8/torch/_dynamo/guards.py#L983